docs/docs/development/hardware-integration/pinctrl.mdx

```diff
@@ -0,0 +1,336 @@
+```dts title="<board>.dts"
+&spi0 {
+    compatible = "nordic,nrf-spim";
+    pinctrl-0 = <&spi0_default>;
+    pinctrl-1 = <&spi0_sleep>;
+    pinctrl-names = "default", "sleep";
+};
+```
+</TabItem>
+<TabItem value="rp2040">
+```dts title="<board>.dts"
+&spi0 {
+    pinctrl-0 = <&spi0_default>;
+    pinctrl-names = "default";
+};
+```
+</TabItem>
+</Tabs>
+This assigns the pins defined in the previous section's examples to the `spi0` node.
+
+Notice that the nRF52840 assigns two items. This is because nodes making use of pin control come with two states by default (though they can have more), a `default` state and a `sleep` state. The nRF52840 can put pins into a "low power state", to reduce power consumption while on sleep. If the RP2040 node made use of pullup or pulldown resistors which had a risk of power leakage while asleep, then it would also define an additional `pinctrl` child node and assign it like in the nRF52840 example.
+
+The nRF52840 example also changes the `compatible` assignment to use SPIM rather than SPI, since it is taking on the "main" role. Check the datasheet for more information about SPIM. The RP2040 makes no such distinction.
+
+### Alias
+
+You may wish to provide an alias to the node for various reasons:
+
+- Compatibility with other boards, if defining for a shield
+- Compatibility with an interconnect
+- Easier personal use
+
+Aliases are assigned like so:
+
+```dts
+    my_alias_spi: &spi0 {};
+```
+
+### Usage
+
+Once you have defined your node, you make use of it by further adjusting the node. You will most likely need to enable the node, as most nodes come disabled:
+
+```dts
+&spi0 {
+    status = "okay";
+};
+```
+
+You would then want to make any adjustments to the node that are necessary, for example adjusting the clock speed. See the Zephyr API documentation for your `compatible` property to see the available properties for customisation. It is recommended to read through the [description of important properties](https://docs.zephyrproject.org/3.5.0/build/dts/intro-syntax-structure.html#dt-important-props), potentially with the addition of [this blog post](https://interrupt.memfault.com/blog/practical_zephyr_dt#zephyrs-dts-skeleton-and-addressing) if `#address-cells` is confusing you.
+
+For SPI specifically, you would create a child node within your SPI bus for each device making use of the SPI bus.
+
+```dts
+&spi0 {
+    cs-gpios = <&gpio0 15 GPIO_ACTIVE_LOW>, <&gpio0 17 GPIO_ACTIVE_LOW>;
+    device1: device@0 {
+        compatible = "manufacturer,device";
+        reg = <0>;
+        spi-max-frequency = <1000000>; /* conservatively set to 1MHz */
+    };
+    device2: device@1 {
+        compatible = "manufacturer,device";
+        reg = <1>;
+        spi-max-frequency = <1000000>; /* conservatively set to 1MHz */
+    };
+};
+```
+
+Additional information on configuring specific SPI buses can be found in other pages of the ZMK documentation, or in the Zephyr documentation.
```

Comment:
"other pages of the ZMK documentation" is this true? 😄 I can't think of a page off the top of my head where this is documented.

Reply:
Maybe you have the shift register docs in mind?

Reply:
So in general, the content on this page is focused on setting up pinctrl and in doing so the bus for devices such as shift registers, RGB, displays, etc. The content on setting up the child node of the bus should be in their respective pages, which is what I'm referring to here. I'll rephrase to make it a bit clearer.

Reply:
That makes sense, OK.